src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+            _infoRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Info);
+            _warningRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Warning);
+            _errorRemoveBracesDiagnosticDescriptor = CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity.Error);
+
+            _severityToRemoveBracesDescriptorMap =
+                new Dictionary<DiagnosticSeverity, DiagnosticDescriptor>
+                {
+                    {DiagnosticSeverity.Hidden, _noneRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Info, _infoRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Warning, _warningRemoveBracesDiagnosticDescriptor },
+                    {DiagnosticSeverity.Error, _errorRemoveBracesDiagnosticDescriptor },
+                };
+        }
+
+        private DiagnosticDescriptor CreateAddBracesDiagnosticDescriptor(DiagnosticSeverity severity) =>
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.AddBracesDiagnosticId,
+                s_addBracesLocalizableTitle,
+                s_addBracesLocalizableMessage,
+                DiagnosticCategory.Style,
+                severity,
+                isEnabledByDefault: true);
+
+
+        private DiagnosticDescriptor CreateRemoveBracesDiagnosticDescriptor(DiagnosticSeverity severity) =>
+            new DiagnosticDescriptor(
+                IDEDiagnosticIds.RemoveBracesDiagnosticId,
+                s_removeBracesLocalizableTitle,
+                s_removeBracesLocalizableMessage,
+                DiagnosticCategory.Style,
+                severity,
+                isEnabledByDefault: true);
+
+        public bool RunInProcess => false;
+
+        public override void Initialize(AnalysisContext context)
+        {
+            var syntaxKindsOfInterest =
+            ImmutableArray.Create(SyntaxKind.IfStatement,
+                SyntaxKind.ElseClause,
+                SyntaxKind.ForStatement,
+                SyntaxKind.ForEachStatement,
+                SyntaxKind.ForEachComponentStatement,
+                SyntaxKind.WhileStatement,
+                SyntaxKind.DoStatement,
+                SyntaxKind.UsingStatement,
+                SyntaxKind.LockStatement,
+                SyntaxKind.FixedStatement);
+
+            context.RegisterSyntaxNodeAction(AnalyzeNode, syntaxKindsOfInterest);
+        }
+
+        public DiagnosticAnalyzerCategory GetAnalyzerCategory() => DiagnosticAnalyzerCategory.SemanticSpanAnalysis;
+
+        public void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var optionSet = context.Options.GetOptionSet();
+            var useBracesOption = optionSet.GetOption(CSharpCodeStyleOptions.UseBracesWherePossible);
+            var severity = useBracesOption.Notification.Value;
+
+            if (useBracesOption.Value)
```

Comment:
In my experience, Hidden doesn't notify, but when you use that severity level, it is also available.